widgetry/src/widgets/button.rs

```diff
@@ -271,6 +308,460 @@ impl Btn {
+        let mut image = self.default_style.image.take().unwrap_or_default();
+        image.content_mode = content_mode;
+        self.default_style.image = Some(image);
+        self
+    }
+
+    fn style_mut(&'b mut self, state: ButtonState) -> &'b mut ButtonStyle<'a> {
+        match state {
+            ButtonState::Default => &mut self.default_style,
+            ButtonState::Hover => &mut self.hover_style,
+            ButtonState::Disabled => &mut self.disable_style,
+        }
+    }
+
+    fn style(&'b self, state: ButtonState) -> &'b ButtonStyle<'a> {
+        match state {
+            ButtonState::Default => &self.default_style,
+            ButtonState::Hover => &self.hover_style,
+            ButtonState::Disabled => &self.disable_style,
+        }
+    }
+
+    pub fn bg_color(mut self, color: Color, for_state: ButtonState) -> Self {
+        self.style_mut(for_state).bg_color = Some(color);
+        self
+    }
+
+    pub fn outline(mut self, thickness: f64, color: Color, for_state: ButtonState) -> Self {
+        self.style_mut(for_state).outline = Some((thickness, color));
+        self
+    }
+
+    pub fn hotkey<MK: Into<MultiKey>>(mut self, key: MK) -> Self {
+        self.hotkey = Some(key.into());
+        self
+    }
+
+    pub fn tooltip(mut self, tooltip: Text) -> Self {
+        self.tooltip = Some(tooltip);
+        self
+    }
+
+    pub fn no_tooltip(mut self) -> Self {
+        // otherwise the widgets `name` is used
+        self.tooltip = Some(Text::new());
+        self
+    }
+
+    pub fn vertical(mut self) -> Self {
+        self.stack_axis = Some(geom_batch_stack::Axis::Vertical);
+        self
+    }
+
+    pub fn horizontal(mut self) -> Self {
+        self.stack_axis = Some(geom_batch_stack::Axis::Horizontal);
+        self
+    }
+
+    pub fn disabled(mut self) -> Self {
+        self.is_disabled = true;
+        self
+    }
+
+    pub fn label_first(mut self) -> Self {
+        self.is_label_before_image = true;
+        self
+    }
+
+    pub fn image_first(mut self) -> Self {
+        self.is_label_before_image = false;
+        self
+    }
+
+    /// Spacing between the image and text of a button.
+    /// Has no effect if the button is text-only or image-only.
+    pub fn stack_spacing(mut self, value: f64) -> Self {
+        self.stack_spacing = value;
+        self
+    }
+
+    // Specific UI treatments
+
+    pub fn dropdown(self) -> Self {
+        self.image_path("system/assets/tools/arrow_drop_down.svg")
+            .image_dims(12.0)
+            .stack_spacing(12.0)
+            .label_first()
+    }
+
+    /// Shorthand method to build a Button wrapped in a Widget
+    pub fn build_widget(&self, ctx: &EventCtx, action: &str) -> Widget {
+        Widget::new(Box::new(self.build(ctx, action))).named(action)
+    }
+
+    /// Shorthand method to build a widget whose action is derived from the label's text.
+    // Does `def` stand for anything meaningful? Is there a better short name?
```

Comment:
"default"
Thinking about it again, I still don't know how to name this better. It feels like a case where method overloading would be nice -- either `.build(ctx)` or `.build(ctx, "label")`

Reply:
Things I miss most from Swift:
- named params (optionally)
- `guard`
- method overloading
- `FooLiteral` traits allowing more liberal usage of literals (IMO `let  f: f32 = 0;` should not explode).